src/charge_controller/ChargeController.cpp

```diff
@@ -88,55 +110,58 @@ void ChargeController::connectedState() {
  * state
  */
 void ChargeController::chargingState() {
-  if (changedState) {
-    relay.writePin(RELAY_ON);
-    changedState = false;
-  }
-  if (!bms.isConnected()) {
-    changedState = true;
-    state = ControllerStates::NO_BATTERY;
-  } else if (!checkBMS()) {
-    changedState = true;
-    state = ControllerStates::FAULT;
-  }
+    if (changedState) {
+        log::LOGGER.log(log::Logger::LogLevel::DEBUG, "Changed state->Charging State");
+        relay.writePin(RELAY_ON);
+        changedState = false;
+    }
+    if (bms.numConnected() == 0) {
+        changedState = true;
+        state = ControllerStates::NO_BATTERY;
+    } else if (checkBMS() != 0) {
+        changedState = true;
+        state = ControllerStates::FAULT;
+    }
 }
 
 /**
  * standby state
  * wait for user input to start charging as long as the BMS is ok
  */
 void ChargeController::standbyState() {
-  if (changedState) {
-    relay.writePin(RELAY_OFF);
-    changedState = false;
-  }
+    if (changedState) {
+        log::LOGGER.log(log::Logger::LogLevel::DEBUG, "Changed state->Standby State, waiting for button");
+        relay.writePin(RELAY_OFF);
+        changedState = false;
+    }
 
-  if (!bms.isConnected()) {
-    changedState = true;
-    state = ControllerStates::NO_BATTERY;
-  } else if (!checkBMS()) {
-    changedState = true;
-    state = ControllerStates::FAULT;
-  }
+    if (bms.numConnected() == 0) {
+        changedState = true;
+        state = ControllerStates::NO_BATTERY;
+    } else if (checkBMS() != 0) {
+        changedState = true;
+        state = ControllerStates::FAULT;
+    }
 }
 
 /**
  * Fault occurred state
  * A fault has occurred, stop charging. When fault is cleared, wait in standby
  */
 void ChargeController::faultState() {
-  if (changedState) {
-    relay.writePin(RELAY_OFF);
-    changedState = false;
-  }
+    if (changedState) {
+        log::LOGGER.log(log::Logger::LogLevel::DEBUG, "Changed state->Fault State");
+        relay.writePin(RELAY_OFF);
+        changedState = false;
+    }
 
-  if (!bms.isConnected()) {
-    changedState = true;
-    state = ControllerStates::NO_BATTERY;
-  } else if (checkBMS()) {
-    changedState = true;
-    state = ControllerStates::STANDBY;
-  }
+    if (bms.numConnected() == 0) {
+        changedState = true;
+        state = ControllerStates::NO_BATTERY;
+    } else if (checkBMS() == 0) {
```

Comment:
Should this be `!checkBMS()`?

Reply:
No this seems correct. If there's no errors that checkBMS is returning, you can put it back in standby mode